components/script/dom/mutationrecord.rs

```diff
@@ -0,0 +1,135 @@
+use core::ptr;
+use dom_struct::dom_struct;
+use dom::bindings::codegen::Bindings::MutationRecordBinding::MutationRecordBinding::MutationRecordMethods;
+use dom::bindings::js::{JS, Root};
+use dom::bindings::str::DOMString;
+use dom::bindings::reflector::Reflector;
+use dom::node::Node;
+use dom::nodelist::NodeList;
+use dom::window::Window;
+use std::default::Default;
+
+#[dom_struct]
+pub struct MutationRecord {
+  reflector_: Reflector,
+
+//  readonly attribute DOMString record_type;
+  record_type: DOMString,
+
+//  [SameObject]
+//  readonly attribute Node target;
+  target: Root<Node>,
```

Comment:
The field in the struct should be `JS<Node>`, but the return value of `fn Target(&self)` should be `Root<Node>`. Use `Root::from_ref(&*self.target)` to get a `Root<T>` from a `JS<T>`.